aws/resource_aws_route53_resolver_rule.go

```diff
@@ -0,0 +1,294 @@
+package aws
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53resolver"
+)
+
+const (
+	route53ResolverRuleStatusDeleted = "DELETED"
+)
+
+func resourceAwsRoute53ResolverRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsRoute53ResolverRuleCreate,
+		Read:   resourceAwsRoute53ResolverRuleRead,
+		Update: resourceAwsRoute53ResolverRuleUpdate,
+		Delete: resourceAwsRoute53ResolverRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"domain_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppressRoute53ZoneNameWithTrailingDot,
```

Comment:
`suppressRoute53ZoneNameWithTrailingDot` https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/diff_suppress_funcs.go#L110-L112

Is suppressing ```domain_name          = "."```

diff_suppress_funcs.go `suppressRoute53ZoneNameWithTrailingDot` should be updated for "." case.